include/heuristic/HeuristicMapper.hpp

```diff
@@ -151,6 +193,99 @@ class HeuristicMapper : public Mapper {
+        singleQubitGateMultiplicity.at(q1)) *
+                         log2(singleQubitFidelities.at(swap.first))
+                      -(singleQubitGateMultiplicity.at(q1) -
+        singleQubitGateMultiplicity.at(q2)) *
+                        log2(singleQubitFidelities.at(swap.second))
+                      );
+        // adding cost oft the swap gate
+        costFixed += (-    log2(twoQubitFidelities.at(source).at(middleAnc))
+                      -    log2(twoQubitFidelities.at(middleAnc).at(target))
+                      -    log2(singleQubitFidelities.at(source))
+                      -    log2(singleQubitFidelities.at(middleAnc))
+                      -2 * log2(singleQubitFidelities.at(target))
+                      -    log2(1.0 -
+        arch.getProperties().readoutErrorRate.get(source))
+                      -    log2(1.0 -
+        arch.getProperties().readoutErrorRate.get(middleAnc)));*/
+      } else {
+        costFixed += COST_TELEPORTATION;
+      }
+    }
+
+    /**
+     * @brief recalculates the fixed cost of the node from current mapping and
+     * swaps
+     *
+     * @param arch the architecture for calculating distances between physical
+     * qubits and supplying qubit information such as fidelity
+     * @param singleQubitGateMultiplicity vector containing the number of gates
+     * acting on each logical qubit
+     * @param considerFidelity controls if qubit fidelity should be taken into
+     * account
+     */
+    void recalculateFixedCost(
+        const Architecture&              arch,
+        const std::vector<std::uint16_t> singleQubitGateMultiplicity,
+        bool                             considerFidelity) {
+      const auto& singleQubitFidelities = arch.getSingleQubitFidelities();
+      const auto& twoQubitFidelities    = arch.getFidelityTable();
+
+      costFixed = 0;
+      if (considerFidelity) {
+        // adding costs of single qubit gates
+        for (std::uint16_t i = 0U; i < arch.getNqubits(); ++i) {
+          if (singleQubitGateMultiplicity.at(i) == 0) {
+            continue;
+          }
+          costFixed += -singleQubitGateMultiplicity.at(i) *
+                       log2(singleQubitFidelities.at(
+                           static_cast<std::size_t>(locations.at(i))));
+        }
+        // adding cost oft the swap gates
+        for (auto& swapNode : swaps) {
+          for (auto& swap : swapNode) {
+            if (swap.op == qc::SWAP) {
+              if (arch.bidirectional()) {
+                costFixed +=
+                    -3 *
+                    log2(twoQubitFidelities.at(swap.first).at(swap.second));
+              } else {
+                costFixed += (-3 * log2(twoQubitFidelities.at(swap.first)
+                                            .at(swap.second)) -
+                              2 * log2(singleQubitFidelities.at(swap.first)) -
+                              2 * log2(singleQubitFidelities.at(swap.second)));
```

Comment:
This looks like Code duplication with the function above. Consider extracting that into a function.